### PR TITLE
Visitor checkout#40

### LINKED ISSUE #40: User Story 31, Visitors must register or log in to check out
```
As a visitor
When I have items in my cart
And I visit my cart
I see information telling me I must register or log in to finish the checkout process
The word "register" is a link to the registration page
The words "log in" is a link to the login page
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #46

depends on #44

depends on #74

depends on #68

depends on #64

### PR DESCRIPTION
**Pull request by Bradniedt**
-Adds test for visitor seeing a message to register or login to checkout. The words register and  log in are links. Added a sad path test to confirm that logged in user does not see this message. 
-Implements functionality to complete user story 31
-All tests passing

closes #40

### CODE REVIEW

File: app/views/carts/show.html.erb [hunk 1]
```diff
@@ -13,3 +13,7 @@
   <% unless @cart.cart_count == 0 %>
     <%= link_to "Empty Cart", cart_path %>
   <% end %>
+
+  <% unless current_user %>
```
Comment (danbriechle) [inline app/views/carts/show.html.erb]: Looks good

**Comment by danbriechle** [comment]
This has my ok for sure !

File: spec/features/visitor_checkout_spec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
+require 'rails_helper'
+
+describe 'As a visitor' do
+  describe 'when I add items to my cart and I visit my cart page' do
+    it 'should see a message telling me to register or login to checkout' do
+      item_1 = create(:item)
+      item_2 = create(:item)
```
Comment (asmolentzov) [inline spec/features/visitor_checkout_spec.rb]: Super small but I don't think you really need item_2?

File: spec/features/visitor_checkout_spec.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      expect(current_path).to eq(registration_path)
+
+      visit cart_path
+
+      click_on 'log in'
+      expect(current_path).to eq(login_path)
+    end
+    it 'should not show the register or login message if I am logged in' do
+      item_1 = create(:item)
+      item_2 = create(:item)
```
Comment (asmolentzov) [inline spec/features/visitor_checkout_spec.rb]: Same here - item_2 unnecessary?

File: spec/features/visitor_checkout_spec.rb (hunk 2 above)
Comment (saleratus) [reply spec/features/visitor_checkout_spec.rb]: I'm kind of with  both of you on item_2. You may be strictly right, Anna, but sometimes it just doesn't feel right testing a list with just one item!

File: app/views/carts/show.html.erb (hunk 1 above)
Comment (saleratus) [reply app/views/carts/show.html.erb]: Nice!

File: spec/features/visitor_checkout_spec.rb (hunk 2 above)
Comment (Bradniedt) [reply spec/features/visitor_checkout_spec.rb]: @asmolentzov and @saleratus  - I see what you mean - I just included it to make sure that item 2 didn't show up. Not super sad-path-y but let me know if you want me to take that out.

Comment (asmolentzov) [reply spec/features/visitor_checkout_spec.rb]: I'm fine either way, but I don't see any spot where you're testing that item_2 doesn't show up when it shouldn't? Or that that is super necessary here, though I'm always for double-checking. Or is it just so that having another item on the page (presumably) doesn't mess things up somehow?

Comment (Bradniedt) [reply spec/features/visitor_checkout_spec.rb]: @asmolentzov I see what you mean now - I'm taking this item_2 out as well as the other. Thanks! =]
